custom_components/octopus_intelligent/octopus_intelligent_system.py

```diff
@@ -33,30 +34,110 @@ def __init__(self, hass, *, api_key, account_id, off_peak_start, off_peak_end):
-
+        self._last_seen_planned_dispatch_source = ""
 
     @property
     def account_id(self):
         return self._account_id
 
-    async def _async_update_data(self):
+    @override
+    async def _async_update_data(self) -> dict[str, Any]:
         """Fetch data from API endpoint.
 
         This is the place to pre-process the data to lookup tables
         so entities can quickly look up their data.
+
+        Returns:
+            dict: The data received from the Octopus API, for example:
+            {
+                'completedDispatches': [{
+                    'chargeKwh': '-0.58',
+                    'startDtUtc': '2024-02-25 02:00:00+00:00',
+                    'endDtUtc': '2024-02-25 02:30:00+00:00',
+                    'meta': {'location': 'AT_HOME', 'source': None},
+                }, {
+                    'chargeKwh': '-0.58',
+                    'startDtUtc': '2024-02-25 03:30:00+00:00',
+                    'endDtUtc': '2024-02-25 04:00:00+00:00',
+                    'meta': {'location': 'AT_HOME', 'source': None},
+                }],
+                'plannedDispatches': [{
+                    'chargeKwh': '-0.67',
+                    'startDtUtc': '2024-02-25 23:30:00+00:00',
+                    'endDtUtc': '2024-02-26 00:00:00+00:00',
+                    'meta': {'location': None, 'source': 'smart-charge'},
+                }, {
+                    'chargeKwh': '-1.12',
+                    'startDtUtc': '2024-02-26 03:00:00+00:00',
+                    'endDtUtc': '2024-02-26 04:00:00+00:00',
+                    'meta': {'location': None, 'source': 'smart-charge'},
+                }],
+                'registeredKrakenflexDevice': {
+                    'chargePointMake': 'No charger (3-pin plug)',
+                    'chargePointModel': '3-pin plug',
+                    'chargePointPowerInKw': '2.400',
+                    'createdAt': '2024-02-08T10:43:34.061834+00:00',
+                    'hasToken': True,
+                    'krakenflexDeviceId': '...',
+                    'provider': 'ENODE',
+                    'status': 'Live',
+                    'suspended': False,
+                    'vehicleBatterySizeInKwh': '58.00',
+                    'vehicleMake': 'Volkswagen',
+                    'vehicleModel': 'ID.3 Pro',
+                },
+                'vehicleChargingPreferences': {
+                    'weekdayTargetSoc': 80,
+                    'weekdayTargetTime': '08:00',
+                    'weekendTargetSoc': 80,
+                    'weekendTargetTime': '08:00',
+                },
+            }
         """
         try:
             # Note: asyncio.TimeoutError and aiohttp.ClientError are already
             # handled by the data update coordinator.
-            async with async_timeout.timeout(90):
-                return await self.client.async_get_combined_state(self._account_id)
+            async with asyncio.timeout(90):
```

Comment:
Incidentally replacing `async_timeout.timeout()` with `asyncio.timeout()` because the former is a third-party PyPi dependency (not declared in `manifest.json`, working “by chance” because some other common HASS integration uses it too) and the latter is a built-in Python library implementation.